src/Ens.ts

```diff
@@ -0,0 +1,594 @@
+      FetchProvider.factory(NamingServiceName.ENS, this.url);
+    this.proxyServiceApiKey = finalSource['proxyServiceApiKey'];
+
+    const registryAddress =
+      finalSource['registryAddress'] || EnsNetworkMap[this.network];
+    this.registryContract = new EthereumContract(
+      ensInterface,
+      registryAddress,
+      this.provider,
+      this.proxyServiceApiKey,
+    );
+
+    const nameWrapperAddress = this.determineNameWrapperAddress(this.network);
+    this.nameWrapperContract = new EthereumContract(
+      nameWrapperInterface,
+      nameWrapperAddress,
+      this.provider,
+      this.proxyServiceApiKey,
+    );
+
+    const baseRegistrarAddress = this.determineBaseRegistrarAddress(
+      this.network,
+    );
+    this.baseRegistrarContract = new EthereumContract(
+      baseRegistrarInterface,
+      baseRegistrarAddress,
+      this.provider,
+      this.proxyServiceApiKey,
+    );
+  }
+
+  static async autoNetwork(
+    config: {url: string} | {provider: Provider},
+  ): Promise<Ens> {
+    let provider: Provider;
+
+    if (hasProvider(config)) {
+      provider = config.provider;
+    } else {
+      if (!config.url) {
+        throw new ConfigurationError(ConfigurationErrorCode.UnspecifiedUrl, {
+          method: NamingServiceName.ENS,
+        });
+      }
+      provider = FetchProvider.factory(NamingServiceName.ENS, config.url);
+    }
+
+    const networkId = (await provider.request({
+      method: 'net_version',
+    })) as number;
+    const networkName = EthereumNetworksInverted[networkId];
+    if (!networkName || !EnsSupportedNetwork.guard(networkName)) {
+      throw new ConfigurationError(ConfigurationErrorCode.UnsupportedNetwork, {
+        method: NamingServiceName.ENS,
+      });
+    }
+    return new this({network: networkName, provider: provider});
+  }
+
+  serviceName(): NamingServiceName {
+    return this.name;
+  }
+
+  namehash(domain: string): string {
+    if (!this.checkSupportedDomain(domain)) {
+      throw new ResolutionError(ResolutionErrorCode.UnsupportedDomain, {
+        domain,
+      });
+    }
+    return eip137Namehash(domain);
+  }
+
+  childhash(parentHash: string, label: string): string {
+    return eip137Childhash(parentHash, label);
+  }
+
+  async isSupportedDomain(domain: string): Promise<boolean> {
+    return this.checkSupportedDomain(domain);
+  }
+
+  async owner(domain: string): Promise<string> {
+    const namehash = this.namehash(domain);
+    return await this.callMethod(this.registryContract, 'owner', [namehash]);
+  }
+
+  async resolver(domain: string): Promise<string> {
+    const nodeHash = this.namehash(domain);
+    const resolverAddr = await this.callMethod(
+      this.registryContract,
+      'resolver',
+      [nodeHash],
+    );
+    if (isNullAddress(resolverAddr)) {
+      throw new ResolutionError(ResolutionErrorCode.UnspecifiedResolver);
+    }
+    return resolverAddr;
+  }
+
+  async record(domain: string, key: string): Promise<string> {
```

Comment:
this should return text record

Reply:
so probably you can call directly the text record API instead of calling all records?

Reply:
Where's the api defined? in what repo?

Reply:
There's this:

```
  private async getTextRecord(domain, key): Promise<string | undefined> {
    const nodeHash = this.namehash(domain);
    const resolver = await this.getResolverContract(domain);
    return await this.callMethod(resolver, 'text', [nodeHash, key]);
  }

```